controllers/mutation_reconcile_looper.go

```diff
@@ -758,3 +765,112 @@ func (m *MutationReconcileLooper) getComponentVersion(ctx context.Context, obj *
 	}
 	return cv, nil
 }
+
+// getValues returns values that can be used for the configuration
+// currently it only possible to use inline values OR values from an external source
+func (m *MutationReconcileLooper) getValues(ctx context.Context, obj *v1alpha1.MutationSpec) (*apiextensionsv1.JSON, error) {
+	if obj.Values != nil {
+		return obj.Values, nil
+	}
+
+	if obj.ValuesFrom.FluxSource != nil {
+		source, err := m.getSource(ctx, obj.ValuesFrom.FluxSource.SourceRef)
+		if err != nil {
+			return nil, fmt.Errorf("could not get values from source: %w", err)
+		}
+
+		tmpDir, err := os.MkdirTemp("", "mutation-controller-")
+		tarSize := tar.UnlimitedUntarSize
+		fetcher := fetch.NewArchiveFetcher(10, tarSize, tarSize, "")
+		err = fetcher.Fetch(source.GetArtifact().URL, source.GetArtifact().Digest, tmpDir)
+		if err != nil {
+			return nil, fmt.Errorf("could not fetch values artifact from source: %w", err)
+		}
+
+		path, err := securejoin.SecureJoin(tmpDir, obj.ValuesFrom.FluxSource.Path)
+		if err != nil {
+			return nil, fmt.Errorf("could not construct values file path: %w", err)
+		}
+
+		dataFile, err := os.ReadFile(path)
+		if err != nil {
+			return nil, fmt.Errorf("could not read values file: %w", err)
+		}
+
+		var data map[string]any
+		if err := yaml.Unmarshal(dataFile, &data); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal values: %w", err)
+		}
+
+		var found bool
+		if obj.ValuesFrom.FluxSource.SubPath != "" {
+			data, found = extractSubpath(data, obj.ValuesFrom.FluxSource.SubPath)
+			if !found {
+				return nil, errors.New("subPath not found")
+			}
+		}
+
+		jsonData, err := json.Marshal(data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal values: %w", err)
+		}
+
+		return &apiextensionsv1.JSON{
+			Raw: jsonData,
+		}, nil
+	}
+
+	return nil, errors.New("no values found")
+}
+
+// Recursive function to extract the subpath from the data map
+func extractSubpath(data map[string]any, subpath string) (map[string]any, bool) {
+	keys := splitSubpath(subpath)
+	curr := data
+
+	for i, key := range keys {
+		value, ok := curr[key]
+		if !ok {
+			return nil, false
+		}
+
+		if i == len(keys)-1 {
+			return value.(map[string]any), true
+		}
+
+		if nested, ok := value.(map[any]any); ok {
+			curr = convertMap(nested)
+		} else {
+			return nil, false
+		}
+	}
+
+	return nil, false
+}
+
+// Helper function to split the subpath into keys
+func splitSubpath(subpath string) []string {
+	return strings.Split(subpath, ".")
+}
+
+func convertMap(data map[interface{}]interface{}) map[string]interface{} {
+	result := make(map[string]interface{})
+
+	for k, v := range data {
+		key, ok := k.(string)
+		if !ok {
+			// Handle the case where the key is not a string
+			continue
+		}
+
+		switch value := v.(type) {
+		case map[interface{}]interface{}:
+			// Recursively convert nested maps
+			result[key] = convertMap(value)
+		default:
+			result[key] = value
+		}
+	}
+
+	return result
+}
```

Comment:
```suggestion
func convertMap(data map[any]any) map[string]any {
	result := make(map[string]any)

	for k, v := range data {
		key, ok := k.(string)
		if !ok {
			// Handle the case where the key is not a string
			continue
		}

		switch value := v.(type) {
		case map[any]any:
			// Recursively convert nested maps
			result[key] = convertMap(value)
		default:
			result[key] = value
		}
	}

	return result
}
```

Just looks neater. :D